native/src/widget/text.rs

```diff
@@ -186,9 +208,33 @@ pub fn draw<Renderer>(
         alignment::Vertical::Bottom => bounds.y + bounds.height,
     };
 
+    let size = size.unwrap_or(renderer.default_size());
+
+    for &Highlight { start, end, color } in highlights {
+        let width_before_start =
+            renderer.measure_width(&content[..start], size, font.clone());
+
+        let width =
+            renderer.measure_width(&content[start..end], size, font.clone());
+
+        let quad = renderer::Quad {
+            bounds: Rectangle {
+                x: x + width_before_start,
+                y,
```

Comment:
It seems highlighting will not work with a `Text` of multiple lines?

Reply:
Yeah, I should clarify that this a very basic and naive implementation thus far; there's a lot of trivial cases that break it.